Core.Collectors.Tests/Web/FixedHttpClient.cs

```diff
@@ -75,19 +75,30 @@ public Task<HttpResponseMessage> GetAsync(string requestUrl, IAuthentication aut
 
         public Task<HttpResponseMessage> GetAsync(string requestUrl, IAuthentication authentication, ProductInfoHeaderValue productInfoHeaderValue)
         {
-            return this.GetAsync(requestUrl, authentication);
+            return this.GetAsync(requestUrl, authentication, productInfoHeaderValue);
         }
 
         public Task<HttpResponseMessage> GetAsync(string requestUrl, IAuthentication authentication, ProductInfoHeaderValue productInfoHeaderValue, string eTag)
         {
-            return this.GetAsync(requestUrl, authentication);
+            return this.GetAsync(requestUrl, authentication, productInfoHeaderValue);
         }
 
         public Task<HttpResponseMessage> PostAsync(string requestUrl, string requestBody, IAuthentication authentication, ProductInfoHeaderValue productInfoHeaderValue)
         {
             return this.PostAsync(requestUrl, authentication, requestBody);
         }
 
+        public Task<HttpResponseMessage> GetAsync(string requestUrl, IAuthentication authentication, ProductInfoHeaderValue productInfoHeaderValue, Dictionary<string, string> additionalHeaders)
```

Comment:
Curiosity question for my education - is your coding guideline discouraging using optional parameters? Looks we've a lot override here the code can be cleaner bringing in an optional param but there's benefit this way making it crisp what exactly must be passed in :)

Reply:
No generic guideline. I prefer to be super explicit in general to make things more difficult to make mistake later on, rather than saving a few lines of code. Some of the explicitness that you see will be based on my personal experience.